visualization/app/codeCharta/util/scenarioHelper.ts

```diff
@@ -49,14 +49,24 @@ export class ScenarioHelper {
 		return scenarioItems
 	}
 
-	private static isScenarioAppliable(scenario: RecursivePartial<Scenario>, metricData: NodeMetricData[]) {
-		if (scenario.area && !metricData.find(x => x.name === scenario.area.areaMetric)) {
-			return false
-		}
-		if (scenario.color && !metricData.find(x => x.name === scenario.color.colorMetric)) {
-			return false
+	private static isScenarioAppliable(scenario: RecursivePartial<Scenario>, metricData: MetricData) {
+		const { area, color, height, edge } = scenario
+
+		if (area || color || height) {
+			const nodeMetricSet = new Set(metricData.nodeMetricData.map(data => data.name))
+
+			if (area && !nodeMetricSet.has(area.areaMetric)) {
+				return false
+			}
+			if (color && !nodeMetricSet.has(color.colorMetric)) {
+				return false
+			}
+			if (height && !nodeMetricSet.has(height.heightMetric)) {
+				return false
+			}
```

Comment:
```suggestion
			if (
				area && !nodeMetricSet.has(area.areaMetric) ||
				color && !nodeMetricSet.has(color.colorMetric) ||
				height && !nodeMetricSet.has(height.heightMetric)
			) {
				return false
			}
```

If we prefer short notations.

Reply:
Oh I like that short notation of the if statements